### PR TITLE
Cleanup tendermint code

### PR DESCRIPTION
**Pull request by majecty**
This PR depends on https://github.com/CodeChain-io/codechain/pull/969

Each commit is simple. Please check it commit by commit except for the first commit.

### CODE REVIEW

**Comment by sgkim126** [comment]
@majecty Please resolve conflict.s

File: core/src/consensus/tendermint/mod.rs [hunk 1]
```diff
@@ -880,13 +875,12 @@ impl ConsensusEngine<CodeChainMachine> for Tendermint {
             return false
         }
 
-        let c = match self.client.read().as_ref().and_then(|weak| weak.upgrade()) {
```
Comment (sgkim126) [inline core/src/consensus/tendermint/mod.rs]: It seems that it changes the behavior.
Is it okay?

File: core/src/consensus/tendermint/mod.rs [hunk 2]
```diff
@@ -480,36 +478,33 @@ impl Tendermint {
     }
 
     fn backup(&self) {
-        if let Some(c) = self.client.read().as_ref().and_then(|weak| weak.upgrade()) {
```
Comment (sgkim126) [inline core/src/consensus/tendermint/mod.rs]: It seems that it changes the behavior.
Is it okay?

File: core/src/consensus/tendermint/mod.rs [hunk 3]
```diff
@@ -919,7 +919,6 @@ impl ConsensusEngine<CodeChainMachine> for Tendermint {
         {
             self.signer.write().set(ap, address, password);
         }
-        self.to_step(Step::Propose);
```
Comment (sgkim126) [inline core/src/consensus/tendermint/mod.rs]: If then, when it starts?

File: core/src/consensus/tendermint/mod.rs (hunk 1 above)
Comment (majecty) [reply core/src/consensus/tendermint/mod.rs]: Yes. The client should valid in all places.

File: core/src/consensus/tendermint/mod.rs (hunk 2 above)
Comment (majecty) [reply core/src/consensus/tendermint/mod.rs]: Yes. The client should be valid in all places.

File: core/src/consensus/tendermint/mod.rs (hunk 3 above)
Comment (majecty) [reply core/src/consensus/tendermint/mod.rs]: The tendermint starts when the timer expires.
NetworkExtension for TendermintExtension's `on_initialize` set timer and the timer calls `to_step` function.

**Comment by sgkim126** [comment]
@majecty Would you extract the last commit to another PR. That is not cleanup code.

**Comment by majecty** [comment]
@sgkim126 I removed it.

**Pull request merged**